internal/payments/service.go

```diff
@@ -0,0 +1,218 @@
+	cparams := &stripe.CustomerParams{
+		Email:       stripe.String(params.Email),
+		Name:        stripe.String(params.Name),
+		Description: stripe.String(params.Description),
+	}
+
+	// Set default source (payment instrument)
+	//if params.SourceToken != "" {
+	//	err := cparams.SetSource(params.SourceToken)
+	//	if err != nil {
+	//		return nil, StripeErr.Wrap(err)
+	//	}
+	//}
+
+	// TODO: delete after migrating from test environment
+	err := cparams.SetSource("tok_visa")
+	if err != nil {
+		return nil, err
+	}
+
+	return s.client.Customers.New(cparams)
+}
+
+// GetCustomer retrieves customer object from stripe network
+func (s *StripeService) GetCustomer(ctx context.Context, id string) (*stripe.Customer, error) {
+	return s.client.Customers.Get(id, nil)
+}
+
+// GetCustomerDefaultPaymentMethod retrieves customer default payment method from stripe network
+func (s *StripeService) GetCustomerDefaultPaymentMethod(ctx context.Context, customerID string) (*stripe.PaymentMethod, error) {
+	cus, err := s.client.Customers.Get(customerID, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if cus.DefaultSource == nil {
+		return nil, StripeErr.New("no default payment method attached to customer")
+	}
+
+	return s.client.PaymentMethods.Get(cus.DefaultSource.ID, nil)
+}
+
+// GetCustomerPaymentsMethods retrieves all payments method attached to particular customer
+func (s *StripeService) GetCustomerPaymentsMethods(ctx context.Context, customerID string) ([]*stripe.PaymentMethod, error) {
+	var err error
+
+	pmparams := &stripe.PaymentMethodListParams{}
+	pmparams.Filters.AddFilter("customer", "", customerID)
+	pmparams.Filters.AddFilter("type", "", "card")
+
+	iterator := s.client.PaymentMethods.List(pmparams)
+	if err = iterator.Err(); err != nil {
+		return nil, err
+	}
+
+	var paymentMethods []*stripe.PaymentMethod
+	for iterator.Next() {
+		pm := iterator.PaymentMethod()
+		paymentMethods = append(paymentMethods, pm)
+	}
+
+	return paymentMethods, nil
+}
+
+// GetPaymentMethod retrieve payment method object from stripe network
+func (s *StripeService) GetPaymentMethod(ctx context.Context, id string) (*stripe.PaymentMethod, error) {
+	return s.client.PaymentMethods.Get(id, nil)
+}
+
+// CreateProjectInvoice creates new project invoice on stripe network from input params.
+// Included line items:
+// - Storage
+// - Egress
+// - ObjectsCount
+// Created invoice has AutoAdvance property set to true, so it will be finalized
+// (no further editing) and attempted to be paid in 1 hour after creation
+func (s *StripeService) CreateProjectInvoice(ctx context.Context, params CreateProjectInvoiceParams) (*stripe.Invoice, error) {
+	// create line items
+	_, err := s.client.InvoiceItems.New(&stripe.InvoiceItemParams{
+		Customer:    stripe.String(params.CustomerID),
+		Description: stripe.String("Storage"),
+		Quantity:    stripe.Int64(int64(params.Storage)),
```

Comment:
Will there ever be a chance that `params.Storage` overflow in the conversion here?

Reply:
I don't think anyone can store more than `9,223,372,036,854,775,807` `GB` in the nearest future